File: software/examples/php/ExampleInterrupt.php
<?php

require_once('Tinkerforge/IPConnection.php');
require_once('Tinkerforge/BrickletIO16.php');

use Tinkerforge\IPConnection;
use Tinkerforge\BrickletIO16;

const HOST = 'localhost';
const PORT = 4223;
const UID = 'XYZ'; // Change XYZ to the UID of your IO-16 Bricklet

// Callback function for interrupt callback
function cb_interrupt($port, $interrupt_mask, $value_mask)
{
    echo "Port: $port\n";
    echo "Interrupt Mask: " . sprintf("%08b", $interrupt_mask) . "\n";
    echo "Value Mask: " . sprintf("%08b", $value_mask) . "\n";
    echo "\n";
}

$ipcon = new IPConnection(); // Create IP connection
$io = new BrickletIO16(UID, $ipcon); // Create device object

$ipcon->connect(HOST, PORT); // Connect to brickd
// Don't use device before ipcon is connected

// Register interrupt callback to function cb_interrupt
$io->registerCallback(BrickletIO16::CALLBACK_INTERRUPT, 'cb_interrupt');

// Enable interrupt on pin 2 of port A
$io->setPortInterrupt('a', 1 << 2);

echo "Press ctrl+c to exit\n";
$ipcon->dispatchCallbacks(-1); // Dispatch callbacks forever

?>

File: software/examples/perl/example_interrupt.pl
#!/usr/bin/perl

use strict;
use Tinkerforge::IPConnection;
use Tinkerforge::BrickletIO16;

use constant HOST => 'localhost';
use constant PORT => 4223;
use constant UID => 'XYZ'; # Change XYZ to the UID of your IO-16 Bricklet

# Callback subroutine for interrupt callback
sub cb_interrupt
{
    my ($port, $interrupt_mask, $value_mask) = @_;

    print "Port: $port\n";
    print "Interrupt Mask: " . sprintf('%08b', $interrupt_mask) . "\n";
    print "Value Mask: " . sprintf('%08b', $value_mask) . "\n";
    print "\n";
}

my $ipcon = Tinkerforge::IPConnection->new(); # Create IP connection
my $io = Tinkerforge::BrickletIO16->new(&UID, $ipcon); # Create device object

$ipcon->connect(&HOST, &PORT); # Connect to brickd
# Don't use device before ipcon is connected

# Register interrupt callback to subroutine cb_interrupt
$io->register_callback($io->CALLBACK_INTERRUPT, 'cb_interrupt');

# Enable interrupt on pin 2 of port A
$io->set_port_interrupt('a', 1 << 2);

print "Press key to exit\n";
<STDIN>;
$ipcon->disconnect();
